src/sock/socket_session.cpp

```diff
@@ -226,12 +226,14 @@ void socket_session<T>::on_read(const error_code ec, const std::size_t)
 template <class T>
 void socket_session<T>::send(const T msg)
 {
+    send_mutex.lock();
```

Comment:
@codetsunami I'm still getting beast assert failure even with the mutex lock. The lock is released in the below on_write handler.